operator/pkg/init.go

```diff
@@ -36,6 +37,37 @@ type InitOptions struct {
 	Karmada        *operatorv1alpha1.Karmada
 }
 
+// Validate is used to validate the initOptions before creating initJob.
+func (opt *InitOptions) Validate() error {
+	var errs []error
+
+	if len(opt.Name) == 0 || len(opt.Namespace) == 0 {
+		return errors.New("unexpected empty name or namespace")
+	}
+	if len(opt.CrdRemoteURL) > 0 {
+		if _, err := url.Parse(opt.CrdRemoteURL); err != nil {
+			return fmt.Errorf("unexpected invalid crds remote url %s", opt.CrdRemoteURL)
+		}
+	}
+	if !util.IsInCluster(opt.Karmada.Spec.HostCluster) && opt.Karmada.Spec.Components.KarmadaAPIServer.ServiceType != corev1.ServiceTypeNodePort {
```

Comment:
If in the public cloud, it could be type LoadBlancer.

Reply:
Yes, currently, we do not support service of type LB. We haven't figured out how to support the public cloud scenario yet. but this is a good question.

Reply:
In the public cloud, just set the type as LB, and the public cloud LB controller will handle it.

Reply:
Ok, I got it and updated.

Reply:
@jwcesign PTAL

Reply:
Nice